modules/planning/tasks/deciders/path_reuse_decider/path_reuse_decider.cc

```diff
@@ -84,21 +86,32 @@ Status PathReuseDecider::Process(Frame* const frame,
     reference_line_info->set_path_reusable(false);
     ADEBUG << "reusable_path_counter[" << reusable_path_counter_
            << "] total_path_counter[" << total_path_counter_ << "]";
+    ADEBUG << "Stop reusing path when optimization failed on change lane path";
     return Status::OK();
   }
 
-  bool path_reusable = false;
-  if (!frame->current_frame_planned_trajectory().is_replan() &&
-      CheckPathReusable(frame, reference_line_info) &&
-      TrimHistoryPath(frame, reference_line_info)) {
-    ADEBUG << "reuse path";
-    path_reusable = true;
-    ++reusable_path_counter_;  // count reusable path
+  // stop reusing current path:
+  // 1. replan path
+  // 2. collision
+  // 3. failed to trim previous path
+  // 4. speed optimization failed on previous path
+  // const auto history_trajectory_type = FrameHistory::Instance()
+  //                                          ->Latest()
+  //                                          ->reference_line_info()
+  //                                          .front()
+  //                                          .trajectory_type();
+  if (path_reusable_) {
+    if (!frame->current_frame_planned_trajectory().is_replan() &&
+        CheckPathReusable(frame, reference_line_info) &&
```

Comment:
replace the name `CheckPathReusable` to avoid confusion, it look like we are expecting false here, util I saw in implementation that you are just returning the `IsCollisionFree` results.